graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/InputObjectMapper.kt

```diff
@@ -32,11 +32,25 @@ import kotlin.reflect.full.primaryConstructor
 import kotlin.reflect.jvm.javaType
 import kotlin.reflect.jvm.jvmErasure
 
+interface InputObjectMapper {
+    /**
+     * SPI intended for other frameworks/libraries that need to customize how input objects are mapped.
+     * Not intended to be used by most users.
+     *
+     * A custom mapper might call the DefaultInputObjectMapper, passing itself to the DefaultInputObjectMapper constructor.
+     * The DefaultInputObjectMapper will invoke the custom mapper each time it goes a level deeper into a nested object structure.
+     * This makes it possible to have a custom mapper that still mostly relies on the default one.
+     * Be careful to not create an infinite recursion calling back and forward though!
+     */
+    fun <T : Any> mapToKotlinObject(inputMap: Map<String, *>, targetClass: KClass<T>): T
```

Comment:
Great feature @paulbakker, my only concern is ...

If I am a Java Application developer that doesn't use Kotlin in my project explicitly why would I need to be concerned on `mapToKotlinObject`.

Reply:
Maybe we could have a `JavaInputObjectMapper` as well as a `KotlinInputObjectMapper`?

Reply:
Since this isn't really targeted at application developers (it's really for library/framework integration), I think anyone who would implement this probably *should* care.